File: week4/q8.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None
head = None
def push(data):
    global head
    new_node = Node(data)
    new_node.next = head
    head = new_node
def pairwise_swap():
    temp = head
    while temp and temp.next:
        temp.data, temp.next.data = temp.next.data, temp.data
        temp = temp.next.next
def display():
    temp = head
    while temp:
        print(temp.data, end=" ")
        temp = temp.next
push(5); push(4); push(3); push(2); push(1)
pairwise_swap()
display()

File: week4_C/q8.c
#include <stdio.h>
#include <stdlib.h>
struct Node { int data; struct Node* next; };
struct Node* head = NULL;
void push(int data) {
    struct Node* newNode = malloc(sizeof(struct Node));
    newNode->data = data;
    newNode->next = head;
    head = newNode;
}
void pairwiseSwap() {
    struct Node* temp = head;
    while (temp && temp->next) {
        int d = temp->data;
        temp->data = temp->next->data;
        temp->next->data = d;
        temp = temp->next->next;
    }
}
void display() {
    struct Node* temp = head;
    while (temp) { printf("%d ", temp->data); temp = temp->next; }
}
int main() {
    push(5); push(4); push(3); push(2); push(1);
    pairwiseSwap();
    display();
